cmd/exporters/influxdb/influxdb.go

```diff
@@ -51,66 +52,82 @@ func (e *InfluxDB) Init() error {
+	e.Logger.Debug().Msgf("using bucket [%s]", *bucket)
 
-	if org = e.Params.GetChildContentS("org"); org == "" {
+	if org = e.Params.Org; org == nil {
 		return errors.New(errors.MISSING_PARAM, "org")
 	}
-	e.Logger.Debug().Msgf("using organization [%s]", org)
+	e.Logger.Debug().Msgf("using organization [%s]", *org)
 
-	if e.token = e.Params.GetChildContentS("token"); e.token == "" {
+	if token = e.Params.Token; token == nil {
 		return errors.New(errors.MISSING_PARAM, "token")
 	} else {
+		e.token = *token
 		e.Logger.Debug().Msg("will use authorization with api token")
 	}
 
-	if v = e.Params.GetChildContentS("version"); v == "" {
-		v = defaultApiVersion
+	if version = e.Params.Version; version == nil {
+		v := defaultApiVersion
+		version = &v
 	}
-	e.Logger.Debug().Msgf("using api version [%s]", v)
+	e.Logger.Debug().Msgf("using api version [%s]", *version)
 
-	if p = e.Params.GetChildContentS("precision"); p == "" {
-		p = defaultApiPrecision
+	if precision = e.Params.Precision; precision == nil {
+		p := defaultApiPrecision
+		precision = &p
 	}
-	e.Logger.Debug().Msgf("using api precision [%s]", p)
+	e.Logger.Debug().Msgf("using api precision [%s]", *precision)
 
 	// user should provide either url or addr
 	// url is expected to be the full write URL with all query params specified (optionally with scheme)
 	// addr is expected to include host only (no scheme, no port)
-	if url = e.Params.GetChildContentS("url"); url == "" {
-		if addr = e.Params.GetChildContentS("addr"); addr == "" {
+	if url = e.Params.Url; url == nil {
+		if addr = e.Params.Addr; addr == nil {
 			return errors.New(errors.MISSING_PARAM, "url or addr")
 		}
 
-		if port = e.Params.GetChildContentS("port"); port == "" {
-			e.Logger.Debug().Msgf("using default port [%s]", defaultPort)
-			port = defaultPort
-		} else if _, err = strconv.Atoi(port); err != nil {
-			return errors.New(errors.INVALID_PARAM, "port")
+		if port = e.Params.Port; port == nil {
+			e.Logger.Debug().Msgf("using default port [%d]", defaultPort)
+			defPort := defaultPort
+			port = &defPort
 		}
 
-		url = "http://" + addr + ":" + port
-		e.url = fmt.Sprintf("%s/api/v%s/write?org=%s&bucket=%s&precision=%s", url, v, org, bucket, p)
+		urlToUSe := "http://" + *addr + ":" + strconv.Itoa(*port)
+		url = &urlToUSe
+		e.url = fmt.Sprintf("%s/api/v%s/write?org=%s&bucket=%s&precision=%s", *url, *version, *org, *bucket, *precision)
 	} else {
-		e.url = url
+		e.url = *url
+		/* Example url: http://localhost:8088/api/v4/write?org=harvest&bucket=harvest&precision=s
+		   step 1: localhost:8088/api/v4/write?org=harvest&bucket=harvest&precision=s
+		   step 2: localhost:8088 value in addrAndPort var
+		   step 3: addr has localhost and port has 8088
+		*/
+		addrAndPort := strings.Split(strings.Split(e.url, "//")[1], "/")[0]
+		res := strings.Split(addrAndPort, ":")
+		addr = &res[0]
+		if p, err := strconv.Atoi(res[1]); err != nil {
+			e.Logger.Warn().Msgf("invalid port [%s]", p)
+		} else {
+			port = &p
```

Comment:
do you need port just for log message later? If yes, why not just log whole url and avoid all this parsing.

Reply:
Agreed. Parsing may not be full proof and url would be more better. 
Url is already logged, so not needed to log again.